File: Level-1/4.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer;
        ArrayList<Integer> list = new ArrayList<>();
        for(int i = 0; i < arr.length; i++){
            if(i == 0) {
                list.add(arr[i]);
            }else if(arr[i - 1] != arr[i]){
                list.add(arr[i]);
            }
        }
        
        answer = new int[list.size()];
        
        for(int i = 0; i < list.size(); i++){
            answer[i] = list.get(i);
        }

        return answer;
    }
}

File: Level-1/4.js
function solution(arr) {
    let answer = [arr[0]];
    for(let i = 1; i < arr.length; i++) {
        if(arr[i] !== answer[answer.length - 1]) {
            answer.push(arr[i]);
        }
    }
    return answer;
}

// function solution(arr) {
//     let answer = [];
    
//     arr.forEach((v) => {
//         if(answer.length) {
//             if(answer[answer.length - 1] !== v) answer.push(v);
//         }else {
//             answer.push(v);
//         }
//     });
    
//     return answer;
// }
